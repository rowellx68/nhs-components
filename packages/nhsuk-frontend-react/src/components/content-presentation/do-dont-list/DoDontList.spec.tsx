import React from 'react';
import { it, expect } from 'vitest';
import { render } from '@testing-library/react';
import { composeStory } from '@storybook/react';
import meta, {
  DoList as DoListStory,
  DontList as DontListStory,
} from './DoDontList.stories';

const DoList = composeStory(DoListStory, meta);
const DontList = composeStory(DontListStory, meta);

it('should render the do variant of the DoDontList component', () => {
  const { getByRole, container } = render(<DoList />);

  expect(getByRole('list')).toHaveClass('nhsuk-list nhsuk-list--tick');
  expect(container.querySelectorAll('.nhsuk-icon__tick')).toHaveLength(3);
  expect(container).toMatchSnapshot();
});

it('should render the dont variant of the DoDontList component', () => {
  const { getByRole, container } = render(<DontList />);

  expect(getByRole('list')).toHaveClass('nhsuk-list nhsuk-list--cross');
  expect(container.querySelectorAll('.nhsuk-icon__cross')).toHaveLength(4);
  expect(container).toMatchSnapshot();
});
