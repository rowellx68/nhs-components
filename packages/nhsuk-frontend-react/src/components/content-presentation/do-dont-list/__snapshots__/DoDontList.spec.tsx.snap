// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render the do variant of the DoDontList component 1`] = `
<div>
  <div
    class="nhsuk-do-dont-list"
  >
    <h2
      class="nhsuk-do-dont-list__label"
    >
      Do
    </h2>
    <ul
      class="nhsuk-list nhsuk-list nhsuk-list--tick"
      role="list"
    >
      <li>
        <svg
          aria-hidden="true"
          class="nhsuk-icon nhsuk-icon__tick"
          fill="none"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M18.4 7.8l-8.5 8.4L5.6 12"
            stroke="#007f3b"
            stroke-linecap="round"
            stroke-width="4"
          />
        </svg>
        cover blisters that are likely to burst with a soft plaster or dressing
      </li>
      <li>
        <svg
          aria-hidden="true"
          class="nhsuk-icon nhsuk-icon__tick"
          fill="none"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M18.4 7.8l-8.5 8.4L5.6 12"
            stroke="#007f3b"
            stroke-linecap="round"
            stroke-width="4"
          />
        </svg>
        wash your hands before touching a burst blister
         
      </li>
      <li>
        <svg
          aria-hidden="true"
          class="nhsuk-icon nhsuk-icon__tick"
          fill="none"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M18.4 7.8l-8.5 8.4L5.6 12"
            stroke="#007f3b"
            stroke-linecap="round"
            stroke-width="4"
          />
        </svg>
        allow the fluid in a burst blister to drain before covering it with a plaster or dressing
         
      </li>
    </ul>
  </div>
</div>
`;

exports[`should render the dont variant of the DoDontList component 1`] = `
<div>
  <div
    class="nhsuk-do-dont-list"
  >
    <h2
      class="nhsuk-do-dont-list__label"
    >
      Don't
    </h2>
    <ul
      class="nhsuk-list nhsuk-list nhsuk-list--cross"
      role="list"
    >
      <li>
        <svg
          aria-hidden="true"
          class="nhsuk-icon nhsuk-icon__cross"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M17 18.5c-.4 0-.8-.1-1.1-.4l-10-10c-.6-.6-.6-1.6 0-2.1.6-.6 1.5-.6 2.1 0l10 10c.6.6.6 1.5 0 2.1-.3.3-.6.4-1 .4z"
            fill="#d5281b"
          />
          <path
            d="M7 18.5c-.4 0-.8-.1-1.1-.4-.6-.6-.6-1.5 0-2.1l10-10c.6-.6 1.5-.6 2.1 0 .6.6.6 1.5 0 2.1l-10 10c-.3.3-.6.4-1 .4z"
            fill="#d5281b"
          />
        </svg>
        do not burst a blister yourself
      </li>
      <li>
        <svg
          aria-hidden="true"
          class="nhsuk-icon nhsuk-icon__cross"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M17 18.5c-.4 0-.8-.1-1.1-.4l-10-10c-.6-.6-.6-1.6 0-2.1.6-.6 1.5-.6 2.1 0l10 10c.6.6.6 1.5 0 2.1-.3.3-.6.4-1 .4z"
            fill="#d5281b"
          />
          <path
            d="M7 18.5c-.4 0-.8-.1-1.1-.4-.6-.6-.6-1.5 0-2.1l10-10c.6-.6 1.5-.6 2.1 0 .6.6.6 1.5 0 2.1l-10 10c-.3.3-.6.4-1 .4z"
            fill="#d5281b"
          />
        </svg>
        do not peel the skin off a burst blister
      </li>
      <li>
        <svg
          aria-hidden="true"
          class="nhsuk-icon nhsuk-icon__cross"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M17 18.5c-.4 0-.8-.1-1.1-.4l-10-10c-.6-.6-.6-1.6 0-2.1.6-.6 1.5-.6 2.1 0l10 10c.6.6.6 1.5 0 2.1-.3.3-.6.4-1 .4z"
            fill="#d5281b"
          />
          <path
            d="M7 18.5c-.4 0-.8-.1-1.1-.4-.6-.6-.6-1.5 0-2.1l10-10c.6-.6 1.5-.6 2.1 0 .6.6.6 1.5 0 2.1l-10 10c-.3.3-.6.4-1 .4z"
            fill="#d5281b"
          />
        </svg>
        do not pick at the edges of the remaining skin
      </li>
      <li>
        <svg
          aria-hidden="true"
          class="nhsuk-icon nhsuk-icon__cross"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M17 18.5c-.4 0-.8-.1-1.1-.4l-10-10c-.6-.6-.6-1.6 0-2.1.6-.6 1.5-.6 2.1 0l10 10c.6.6.6 1.5 0 2.1-.3.3-.6.4-1 .4z"
            fill="#d5281b"
          />
          <path
            d="M7 18.5c-.4 0-.8-.1-1.1-.4-.6-.6-.6-1.5 0-2.1l10-10c.6-.6 1.5-.6 2.1 0 .6.6.6 1.5 0 2.1l-10 10c-.3.3-.6.4-1 .4z"
            fill="#d5281b"
          />
        </svg>
        do not wear the shoes or use the equipment that caused your blister until it heals
      </li>
    </ul>
  </div>
</div>
`;
